File: FEM/Stack.ts
class Stack<T> {
    private stack: T[];
    constructor() {
        this.stack = [];
    }

    isEmpty(): boolean {
        return this.stack.length === 0;
    }

    push(item: T): void {
        this.stack.push(item);
    }

    pop(): T | undefined {
        return this.stack.pop();
    }

    peek(): T | undefined {
        return this.stack[this.stack.length-1];
    }

    size(): number {
        return this.stack.length;
    }

    print(): void {
        console.log(this.stack);
    }
}

const st = new Stack<number>();
st.push(1);
st.push(2);
st.push(3);
st.push(4);
st.push(5);
st.print();

console.log(st.pop());

console.log(st.peek());

console.log(st.size());

st.print();

console.log(st.isEmpty());

File: dsa-python/stack.py
class Stack:
    def __init__(self):
        self.items = []

    def is_empty(self):
        return self.items == []
    
    def push(self,item):
        self.items.append(item)

    def pop(self):
        if self.is_empty():
            return "Stack is empty"
        return self.items.pop()
    
    def peek(self):
        if self.is_empty():
            return "Stack is empty"
        return self.items[-1]
    
    def size(self):
        return len(self.items)
    
    def print_stack(self):
        print(self.items)

stacks = Stack()
stacks.push(1)
stacks.push(2)
stacks.push(3)
stacks.push(4)
stacks.push(5)
stacks.print_stack()
stacks.pop()
stacks.print_stack()